File: codes/src-tgt/java-py/HumanEval_9_rolling_max.py
### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static ArrayList<Long> rollingMax(ArrayList<Long> numbers) {
        long running_max = Long.MIN_VALUE;
        ArrayList<Long> result = new ArrayList<Long>();

        for (long n : numbers) {
            running_max = Math.max(running_max, n);
            result.add(running_max);
        }

        return result;
    }
    

}


### Python version

from typing import List, Tuple


def rolling_max(numbers: List[int]) -> List[int]:
    running_max = None
    result = []

    for n in numbers:
        if running_max is None:
            running_max = n
        else:
            running_max = max(running_max, n)

        result.append(running_max)

    return result

File: dpo_train/generation_output/cpp-java/HumanEval_9_rolling_max.java
### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
std::vector<long> rolling_max(std::vector<long> numbers) {
    std::vector<long> result;
    long running_max = 0;
    for (long n : numbers) {
        running_max = std::max(running_max, n);
        result.push_back(running_max);
    }
    return result;
}

### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static ArrayList<Long> rollingMax(ArrayList<Long> numbers) {
        ArrayList<Long> result = new ArrayList<>();
        for (long n : numbers) {
            result.add(n);
        }
        return result;
    }
}

class Solution {
    public static void main(String[] args) throws Exception {
        long[] numbers = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        ArrayList<Long> result = Problem.rollingMax(numbers);
        for (long n : result) {
            System.out.println(n);
        }
    }
}
